### DISCUSSION TITLE
Adding custom elements in json output

### DISCUSSION THREAD

**Discussion by Ganimurugan** [behave/behave#1105] [Q&A] [answered]
We are having the scenario's test status as "passed" in the json output in console or the exported json

            },
            "result": { "status": "passed", "duration": 0.0001163482666015625 }
          }
        ],
        "status": "passed"
      }
    ]

I want to add a new element "logs_captured" below the "status" element which holds trace of logs captured using logging library for a particular scenario. Note: I have the captured traces in "after_scenario" added behave context.
Any leads would be helpful.

Thanks in advance.

**Comment by jenisys** [comment, marked as answer]
The existing JSON schema has no attribute that you can use to add this information.
You need to extend the existing JSON output format/schema to add the additional information.
You need to provide an own JSON formatter to add this additional data to the JSON output.

NOTE: 
`Formatters` are an extension-point in `behave`.
Therefore, you are able to add your own format(ter) class and use it.
You probably need to override `JSONFormatter.finish_current_scenario()` for what you are trying to do.

SEE ALSO:
* [etc/json/behave.json-schema](https://github.com/behave/behave/blob/main/etc/json/behave.json-schema#L75)
* [behave.formatter.json:JSONFormatter](https://github.com/behave/behave/blob/main/behave/formatter/json.py#L25)
* [behave.ini](https://github.com/behave/behave/blob/main/behave.ini#L25) (section `behave.formatters`: Formatter aliases in `behave` config-file to simplify use and/or override existing formatters)
